### PR TITLE
Added parameters for PbPb

### PR DESCRIPTION
**Pull request by cpuggion84**

### CODE REVIEW

**Comment by cpuggion84** [comment]
+async-label async-2023-pbpb-apass4

**Comment by davidrohr** [comment]
is it intended that you set `zdcPbPb.json` independent from the beam type?

**Comment by cpuggion84** [comment]
> is it intended that you set `zdcPbPb.json` independent from the beam type?
You're right, it's better to have a selection.
I didn't put it because the detector typically only takes data in PbPb. But in VDMs it can happen that it also takes data in proton.
Could you please take a look if it's ok? Because I wouldn't know how to try it locally

**Comment by davidrohr** [comment]
it looks reasonable, but I have not tested it.
You can use the feature to run with a local o2dpg checkout to test new o2dpg features.

File: DATA/production/qc-workflow.sh [hunk 1]
```diff
@@ -188,7 +188,13 @@ elif [[ -z ${QC_JSON_FROM_OUTSIDE:-} ]]; then
     [[ -z "${QC_JSON_FV0:-}" ]] && QC_JSON_FV0=$O2DPG_ROOT/DATA/production/qc-async/fv0.json
     [[ -z "${QC_JSON_FDD:-}" ]] && QC_JSON_FDD=$O2DPG_ROOT/DATA/production/qc-async/fdd.json
     [[ -z "${QC_JSON_MID:-}" ]] && QC_JSON_MID=$O2DPG_ROOT/DATA/production/qc-async/mid.json
-    [[ -z "${QC_JSON_ZDC:-}" ]] && has_processing_step ZDC_RECO && QC_JSON_ZDC=$O2DPG_ROOT/DATA/production/qc-async/zdc.json
+    if [[ -z "${QC_JSON_ZDC:-}" ]] && has_processing_step ZDC_RECO; then
+      if [[ "$BEAMTYPE" == "PbPb" ]]; then
+        QQC_JSON_ZDC=$O2DPG_ROOT/DATA/production/qc-async/zdcPbPb.json
```
Comment (JianLIUhep) [inline DATA/production/qc-workflow.sh:193]: There is a typo `QQC`. Please fix it.

**Review by JianLIUhep — CHANGES_REQUESTED** [review]

**Comment by JianLIUhep** [comment]
@cpuggion84 I tested it locally. After fixing the typo, the async workflows worked without crashes and the ZDC QC plots were filled. cc @davidrohr @chiarazampolli

**Comment by davidrohr** [comment]
Merged and fixed typo (https://github.com/AliceO2Group/O2DPG/pull/1775)

**Comment by cpuggion84** [comment]
:cpass0

**Comment by cpuggion84** [comment]
+cpass0

**Comment by alcaliva** [comment]
Dear @cpuggion84, I tried to port this commit to the tag for PbPb (async-v1-01-21) starting from the branch async-v1-01-branch and it gave me conflicts. Please, let me know what is missing and I will include it in the next production tag
Thanks
Alberto

**Comment by aferrero2707** [comment]
@alcaliva I am helping Carlo to figure out what is wrong... we tried to checrry-pick the two commits in the PR into `async-v1-01-branch`, and there were no conflicts. Can you show which conflicts you are getting?
Thanks!

**Comment by alcaliva** [comment]
ERROR: There was a problem cherry-picking 6d96fa8ff5b0da0d9f1ae64b32ceb4773c23a5a8
  Auto-merging DATA/production/qc-workflow.sh
  CONFLICT (content): Merge conflict in DATA/production/qc-workflow.sh
  error: could not apply 6d96fa8f... Update qc-workflow.sh
  hint: After resolving the conflicts, mark them with
  hint: "git add/rm <pathspec>", then run
  hint: "git cherry-pick --continue".
  hint: You can instead skip this commit with "git cherry-pick --skip".
  hint: To abort and get back to the state before "git cherry-pick",
  hint: run "git cherry-pick --abort".

**Comment by chiarazampolli** [comment]
Ciao,

There might other PRs that are needed. We can check tomorrow.

Chiara

**Comment by aferrero2707** [comment]
@alcaliva I still cannot reproduce your error... I have tried to cherry-pick the two commits from master to the `async-v1-01-branch` branch as you are doing, and all goes fine:
```
[flp@mchflp1 O2DPG-async-v1-01]$ git checkout async-v1-01-branch
Switched to branch 'async-v1-01-branch'
[flp@mchflp1 O2DPG-async-v1-01]$ git branch -vv
* async-v1-01-branch 90aa3cd [origin/async-v1-01-branch] Add the exotic injection script for resonances
  master             9946a60 [origin/master] avoid that PRODSPLIT is overwritten
[flp@mchflp1 O2DPG-async-v1-01]$ git cherry-pick ca3d4ce3ee86900393f55e5676fdadc7ec10b89a
[async-v1-01-branch 0084763] Added parameters for PbPb
 Author: Carlo Puggioni <carlo.puggioni@cern.ch>
 3 files changed, 84 insertions(+), 11 deletions(-)
 create mode 100644 DATA/production/qc-async/zdcPbPb.json
[flp@mchflp1 O2DPG-async-v1-01]$ git cherry-pick 6d96fa8ff5b0da0d9f1ae64b32ceb4773c23a5a8
[async-v1-01-branch 4e2a28f] Update qc-workflow.sh
 Author: cpuggion84 <90757182+cpuggion84@users.noreply.github.com>
 1 file changed, 7 insertions(+), 1 deletion(-)
[flp@mchflp1 O2DPG-async-v1-01]$ 
```

Am I doing something wrong?

**Comment by chiarazampolli** [comment]
The PR should be picked on version 12c.

I cannot try today, sorry...

**Comment by alcaliva** [comment]
Hi Chiara, 

I tried yesterday and re-tried now with:

start_from: &start_from async-v1-01-12c
target_tag: &target_tag async-v1-01-12d

I still get conflicts:
INFO: Run cherry-picking in package O2DPG
INFO: Sort commits from oldest to most recent
['8a53dfefad39919f0e4c59535b315762dc123ed6', '6d96fa8ff5b0da0d9f1ae64b32ceb4773c23a5a8']
SUCCESS  8a53dfefad39919f0e4c59535b315762dc123ed6
ERROR: There was a problem cherry-picking 6d96fa8ff5b0da0d9f1ae64b32ceb4773c23a5a8
  Auto-merging DATA/production/qc-workflow.sh
  CONFLICT (content): Merge conflict in DATA/production/qc-workflow.sh
  error: could not apply 6d96fa8f... Update qc-workflow.sh
  hint: After resolving the conflicts, mark them with
  hint: "git add/rm <pathspec>", then run
  hint: "git cherry-pick --continue".
  hint: You can instead skip this commit with "git cherry-pick --skip".
  hint: To abort and get back to the state before "git cherry-pick",
  hint: run "git cherry-pick --abort".
INFO: Trying to continue
INFO: Cherry-picking
  SUCCESS: 1 (8a53dfefad39919f0e4c59535b315762dc123ed6)
  SKIPPED: 0 ()
  FAILED: 1 (6d96fa8ff5b0da0d9f1ae64b32ceb4773c23a5a8)
INFO: Cherry-picking has failed, resetting everything in package O2DPG
==> Log file is at o2dpg_cherry_picks_logs/20241103/run_cherry_pick_tag_1730650686.log.

**Comment by alcaliva** [comment]
Hi @aferrero2707 , 

I see that you are trying to cherry-pick this commit: ca3d4ce3ee86900393f55e5676fdadc7ec10b89a
Can you point me to the PR? Does this have the label async-2024-PbPb-cpass0 ?

**Comment by aferrero2707** [comment]
@alcaliva https://github.com/AliceO2Group/O2DPG/commit/ca3d4ce3ee86900393f55e5676fdadc7ec10b89a is the first of the two commits of this PR.

**Comment by alcaliva** [comment]
Hi Andrea, sorry, from https://aliceo2group.github.io/dpg-documentation/docs/software/requests_automatic/dpg_pr_report_MERGED.html I saw only one. 

It worked now!
The tag is building in https://alijenkins.cern.ch/job/Build%20async%20reco%20O2%20for%20CPU+GPU/197/

**Comment by chiarazampolli** [comment]
Hello @alcaliva , @aferrero2707 , 

I don't know how @aferrero2707  got the second hash, but I see it doing the usual "git log":

```
zampolli@pcalicecz5:~/SOFT/alibuild/ali-o2-dev1/O2DPG$ git log  6d96fa8
commit 6d96fa8ff5b0da0d9f1ae64b32ceb4773c23a5a8
Author: cpuggion84 <90757182+cpuggion84@users.noreply.github.com>
Date:   Mon Oct 21 12:31:14 2024 +0200

    Update qc-workflow.sh

commit ca3d4ce3ee86900393f55e5676fdadc7ec10b89a
Author: Carlo Puggioni <carlo.puggioni@cern.ch>
Date:   Fri Oct 18 16:50:16 2024 +0200

    Added parameters for PbPb

```

@davidrohr , @cpuggion84 , we should squash such commits the next time.

Chiara

**Comment by chiarazampolli** [comment]
Not needed for PbPb 2023 apass4, since the pass is over, and the PR does not affect MC, so the label was removed.

**Comment by chiarazampolli** [comment]
Hello @alcaliva , 

Why did you add again the cpass0 PbPb 2024 label? I think the tag was done with this. Maybe it was a mistake. See also:

https://aliceo2group.github.io/dpg-documentation/docs/software/accepted/O2DPG_accepted.html

or:

https://github.com/AliceO2Group/O2DPG/commits/async-v1-01-12e/

I am removing the label.

Chiara

**Pull request merged**